### PR TITLE
UDP integration

### PR DESCRIPTION
**Pull request by bosr**
# Summary
- [x]  Integrate UDP using git subrepo
- [x] Unit tests

### CODE REVIEW

**Comment by bosr** [comment]
I removed my 'echo' hack as it does not always solve the Travis hanging problem :/

**Comment by bosr** [comment]
rebased

File: Modules/UDP/.gitignore [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Created by https://www.gitignore.io/api/xcode
```
Comment (Danappelxx) [inline Modules/UDP/.gitignore]: could you use the one that SPM generates? https://github.com/Zewo/Zewo/blob/master/.gitignore

File: Modules/UDP/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+notifications:
```
Comment (Danappelxx) [inline Modules/UDP/.travis.yml:1]: Could you replace this with the updated version such as [here](https://github.com/Zewo/Zewo/blob/master/Modules/Core/.travis.yml)? A few minor differences

File: Modules/UDP/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2015 Zewo
```
Comment (Danappelxx) [inline Modules/UDP/LICENSE]: Could you update this to 2016?

File: Modules/UDP/Package.swift [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+let package = Package(
+    name: "UDP",
+    dependencies: [
+        .Package(url: "https://github.com/Zewo/POSIX.git", majorVersion: 0, minor: 13),
+        .Package(url: "https://github.com/Zewo/IP.git", majorVersion: 0, minor: 13),
+
+
+        // test-only dependencies (not yet available in swiftpm: SE-0019)
+        .Package(url: "https://github.com/Zewo/Venice.git", majorVersion: 0, minor: 13),
```
Comment (Danappelxx) [inline Modules/UDP/Package.swift:11]: Hmm, we should find a better way to handle this at some point. @paulofaria

File: Modules/UDP/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+[![Travis][travis-badge]][travis-url]
+
+## Installation
+
+```swift
+import PackageDescription
+
+let package = Package(
+    dependencies: [
+        .Package(url: "https://github.com/VeniceX/UDP.git", majorVersion: 0, minor: 5)
```
Comment (Danappelxx) [inline Modules/UDP/README.md]: `minor: 13`

Comment (Danappelxx) [reply Modules/UDP/README.md]: Also `Zewo/UDP` :)

File: Modules/UDP/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+## Basic usage
+
+### Send messages
+
+Send a message containing "Hello world" from `localhost:5050` to `localhost:5051`
+
+```swift
+import UDP
+
+let udpSocket = try? UDPSocket(ip: IP(port: 5050))
```
Comment (Danappelxx) [inline Modules/UDP/README.md]: I would just use `try` instead of `try` - makes this cleaner and more understandable.

File: Modules/UDP/README.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+The entire Zewo code base is licensed under MIT. By contributing to Zewo you are contributing to an open and engaged community of brilliant Swift programmers. Join us on [Slack](http://slack.zewo.io) to get to know us!
+
+## License
+
+This project is released under the MIT license. See [LICENSE](LICENSE) for details.
+
+[swift-badge]: https://img.shields.io/badge/Swift-3.0-orange.svg?style=flat
+[swift-url]: https://swift.org
+[zewo-badge]: https://img.shields.io/badge/Zewo-0.5-FF7565.svg?style=flat
```
Comment (Danappelxx) [inline Modules/UDP/README.md]: zewo-0.13

File: Modules/UDP/README.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+[zewo-url]: http://zewo.io
+[platform-badge]: https://img.shields.io/badge/Platforms-OS%20X%20--%20Linux-lightgray.svg?style=flat
+[platform-url]: https://swift.org
+[mit-badge]: https://img.shields.io/badge/License-MIT-blue.svg?style=flat
+[mit-url]: https://tldrlegal.com/license/mit-license
+[slack-image]: http://s13.postimg.org/ybwy92ktf/Slack.png
+[slack-badge]: https://zewo-slackin.herokuapp.com/badge.svg
+[slack-url]: http://slack.zewo.io
+[travis-badge]: https://travis-ci.org/VeniceX/UDP.svg?branch=master
+[travis-url]: https://travis-ci.org/VeniceX/UDP
```
Comment (Danappelxx) [inline Modules/UDP/README.md]: Zewo/UDP

**Review by Danappelxx — CHANGES_REQUESTED** [review]
Just a few things! None of them are code related - that stuff looks good :)

**Comment by bosr** [comment]
That's some serious proofreading, I'm impressed. I'll fix this very soon.  

Thanks

**Comment by Danappelxx** [comment]
Hahaha, I've done this sort of thing a few times now so the little typos really jump out at me 😃

**Comment by bosr** [comment]
All done. 

The tests fail partly because of a time tolerance on Ticker Tests that is too strict. See https://github.com/Zewo/Zewo/pull/160

**Review by Danappelxx — APPROVED** [review]

**Comment by Danappelxx** [comment]
Excellent!

**Comment by bosr** [comment]
rebased :)

File: Modules/UDP/Package.swift (hunk 4 above)
Comment (paulofaria) [reply Modules/UDP/Package.swift:11]: I don't think we should do any hacks until SwifPM is fixed.

**Pull request merged**